### PR TITLE
Source connector performance harness

### PR DESCRIPTION
**Pull request by rodireich**
## What
Connector performance command

### CODE REVIEW

**Comment by rodireich** [comment]
/test-connector-performance connector=airbyte/source-postgres

**Comment by rodireich** [comment]
/test-connector-performance connector=airbyte/source-postgres

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=10m

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=vmaltsev/18821-handle-nonstandard-date

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=bottleneck_stream1

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=bottleneck_stream1

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=bottleneck_stream1

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=bottleneck_stream1

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=10m

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=bottleneck_stream1

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=10m

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-mysql ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-mysql ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-mysql ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-mysql ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=10m

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=bottleneck_stream1

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=10m

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cm

**Comment by rodireich** [comment]
/test-connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

File: airbyte-integrations/connectors-performance/source-harness/Dockerfile [hunk 1]
```diff
@@ -0,0 +1,18 @@
+FROM airbyte/integration-base-java:dev AS build
+WORKDIR /airbyte
+ENV APPLICATION source-harness
+COPY build/distributions/${APPLICATION}*.tar ${APPLICATION}.tar
+RUN tar xf ${APPLICATION}.tar --strip-components=1 && rm -rf ${APPLICATION}.tar
+FROM airbyte/integration-base-java:dev
+WORKDIR /airbyte
+ENV APPLICATION source-harness
+COPY --from=build /airbyte /airbyte
+COPY base.sh .
```
Comment (davinchia) [inline airbyte-integrations/connectors-performance/source-harness/Dockerfile:10]: Llooks like there is some duplication here from the integration-base-java image. Is that intentional?

File: airbyte-integrations/connectors-performance/source-harness/Dockerfile [hunk 2]
```diff
@@ -0,0 +1,18 @@
...
+
+RUN curl -O https://s3.us-west-2.amazonaws.com/amazon-eks/1.25.6/2023-01-30/bin/linux/amd64/kubectl
```
Comment (davinchia) [inline airbyte-integrations/connectors-performance/source-harness/Dockerfile:12]: nit: comment why we need kubectl

File: settings.gradle [hunk 3]
```diff
@@ -131,6 +131,9 @@ if (!System.getenv().containsKey("SUB_BUILD") || System.getenv().get("SUB_BUILD"
     include ':airbyte-integrations:connectors:destination-gcs'
 
     include ':tools:code-generator'
+
+    include 'airbyte-integrations:connectors-performance:source-harness'
+    findProject(':airbyte-integrations:connectors-performance:source-harness')?.name = 'source-harness'
```
Comment (davinchia) [inline settings.gradle:136]: what is this `findProject` for?

File: .github/workflows/connector-performance-command.yml [hunk 4]
```diff
@@ -19,15 +19,184 @@ on:
...
+          kubectl logs --follow $POD
+          echo "reading end of log"
+          kubectl logs --tail=5 $POD
+          # RUN_RESULT=$(kubectl logs --tail=5 $POD 2>&1)
+          prefix='{"type":"LOG","log":{"level":"INFO","message":"INFO i.a.i.p.PerformanceTest(runTest):165'
+          suffix='"}}'
+          EOF=$(dd if=/dev/urandom bs=15 count=1 status=none | base64)
+          echo "RUN_RESULT<<$EOF" >> $GITHUB_OUTPUT
+          kubectl logs --tail=1 $POD | while read line ; do line=${line#"$prefix"}; line=${line%"$suffix"}; echo $line >> $GITHUB_OUTPUT ; done 
+          echo "$EOF" >> $GITHUB_OUTPUT
```
Comment (davinchia) [inline .github/workflows/connector-performance-command.yml:160]: Thinking out loud: what about turning this into a script? 

1) This way, someone can run the script locally if needed.
2) The GHA workflow files mostly deal with orchestration logic.

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd dataset=10m

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

File: airbyte-integrations/connectors-performance/source-harness/Dockerfile (hunk 1 above)
Comment (rodireich) [reply airbyte-integrations/connectors-performance/source-harness/Dockerfile]: I rearranged the file, but logically this is doing something similar to what [source-postgres/Dockerfile](https://github.com/airbytehq/airbyte/blob/master/airbyte-integrations/connectors/source-postgres/Dockerfile) is doing for example. 
Just adding a bunch of packages and its own base.sh

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

File: airbyte-integrations/connectors-performance/source-harness/Dockerfile (hunk 2 above)
Comment (rodireich) [reply airbyte-integrations/connectors-performance/source-harness/Dockerfile]: Done. 
Tweaked also build to curl the correct kubectl architecture.
for the arm64 enthusiasts among us

File: settings.gradle (hunk 3 above)
Comment (rodireich) [reply settings.gradle:136]: This was created by IntelliJ's module manager.
Doesn't appear to be necessary. removed
thanks

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

**Comment by rodireich** [comment]
/connector-performance connector=connectors/source-postgres ref=rodi-performance-harness-cmd

File: .github/workflows/connector-performance-command.yml (hunk 4 above)
Comment (rodireich) [reply .github/workflows/connector-performance-command.yml]: I agree with everything - it makes more sense to have this as a separate script.
But if pl let me do this in the next stage: I need to hook the harness to some daily process and collect the result (more on that offline)
For now I sanitized the code and the whole thing is ~10 lines

**Review by davinchia — APPROVED** [review]

**Pull request merged**